### PR TITLE
feat(admin-module): base admin module

### PR DESCRIPTION
**Pull request by brannon-darby**

### CODE REVIEW

File: fireloop/server/model-config.json [hunk 1]
```diff
@@ -1,20 +1,7 @@
...
-    "mixins": [
-      "loopback/common/mixins",
-      "loopback/server/mixins",
-      "../common/mixins",
-      "./mixins",
-      "../node_modules/@mean-expert/loopback-stats-mixin/dist",
-      "../node_modules/@mean-expert/loopback-component-realtime/dist/mixins",
-      "../node_modules/loopback-ds-timestamp-mixin"
-    ]
+    "sources": ["loopback/common/models", "loopback/server/models", "../common/models", "./models"],
```
Comment (beeman) [inline fireloop/server/model-config.json]: Please leave the formatting as is here :)

File: webapp/src/app/admin/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      </div>
+    </app-card>
+  `,
+  styleUrls: ['../admin.component.scss']
+})
+
+export class DashboardComponent implements OnInit {
+
+  dashCards: any = [];
+  // users
```
Comment (beeman) [inline webapp/src/app/admin/dashboard/dashboard.component.ts]: Rogue comments?

File: webapp/src/app/admin/roles/role.component.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+  selector: 'app-role',
+  templateUrl: './role.component.html',
+})
+export class RoleComponent implements OnDestroy {
+
+  private modalRef;
+  // subscriptions
```
Comment (beeman) [inline webapp/src/app/admin/roles/role.component.ts]: Comment is not really informative, might as well go :)

File: webapp/src/app/admin/roles/role.service.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      }),
+      this.formService.input('description', {
+        label: 'Description',
+        className: 'col-12 col-lg-6',
+        addonLeft: {
+          class: 'fa fa-fw fa-comment'
+        }
+      }),
+    ];
+    // console.log(fields);
```
Comment (beeman) [inline webapp/src/app/admin/roles/role.service.ts]: console.log comment

File: webapp/src/app/admin/users/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        className: 'col-12 col-lg-6',
+        addonLeft: {
+          class: 'fa fa-fw fa-user-o'
+        }
+      }),
+    ];
+    if (formType === 'update') {
+      fields.splice(1, 1);
+    }
+    // console.log(fields);
```
Comment (beeman) [inline webapp/src/app/admin/users/user.service.ts]: console.log

File: webapp/src/app/home/admin.module.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+// import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+
+import { SDKBrowserModule } from '../shared/sdk/index';
+import { AdminComponent } from './admin.component';
+import { AdminRoutingModule } from './admin.routing';
+import { UiModule } from '../ui/ui.module';
+
+<<<<<<< HEAD
```
Comment (beeman) [inline webapp/src/app/home/admin.module.ts]: Leftovers from a merge

File: webapp/src/app/home/admin.routing.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { AdminComponent } from './admin.component';
+<<<<<<< HEAD
```
Comment (beeman) [inline webapp/src/app/home/admin.routing.ts]: More merge leftovers

**Review by beeman — CHANGES_REQUESTED** [review]

File: webapp/src/app/admin/admin.module.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import { RoleModule } from './roles/role.module';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    FormsModule,
+    UiModule,
+    UserModule,
+    RoleModule,
+    SDKBrowserModule,
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/admin.module.ts]: Why importing the SDKBrowserModule in here if it is already imported in app.module?

**Review by jonathan-casarrubias — CHANGES_REQUESTED** [review]

File: webapp/src/app/admin/admin.module.ts (hunk 8 above)
Comment (brannon-darby) [reply webapp/src/app/admin/admin.module.ts]: nice catch, @jonathan-casarrubias

Comment (brannon-darby) [reply webapp/src/app/admin/admin.module.ts]: same is true for UiModule, or should that stay?

@beeman @jonathan-casarrubias

Comment (jonathan-casarrubias) [reply webapp/src/app/admin/admin.module.ts]: yup

**Review by jonathan-casarrubias — APPROVED** [review]

**Pull request merged**